File: src/test/java/ex_pure/CountSourceLinesTest.java
package ex_pure;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.net.URISyntaxException;
import java.nio.file.Path;

import static org.junit.jupiter.api.Assertions.assertEquals;

class CountSourceLinesTest {

    @Test
    void countSourceLines() throws URISyntaxException, IOException {
        Path longTextFile = Path.of(ClassLoader.getSystemResource("ex_pure/lipsum.txt").toURI());

        long result = CountSourceLines.countSourceLines(longTextFile);

        assertEquals(120, result);
    }

}

File: src/test/ts/ex_pure/CountSourceLines.test.js
import { expect, test } from 'vitest'
import { countSourceLines } from '../../../main/ts/ex_pure/CountSourceLines.js'

test('count lines in lipsum.txt is 120', () => {
    expect(countSourceLines('src/test/resources/ex_pure/lipsum.txt')).toBe(120)
})
